### ISSUE TITLE
search feature

### ISSUE THREAD

**Issue by adamfoneil** [adamfoneil/MyMusic.Wpf#4]
On the All Files tab toolbar, I'd like to have a right-aligned text box that searches the MP3 database (Title, Artist, Album, and Filename) and displays results in the existing grid.

Would be neat if the search box itself had cue text or placeholder that said "search" along with some kind of X button to clear the search and show all results again.

**Comment by adamfoneil** [comment]
Using [feature/branch](https://github.com/adamfoneil/MyMusic.Wpf/tree/feature/search)

**Comment by adamfoneil** [comment]
Note that I removed the toolbar in order to get the right-aligned search text box I wanted. (couldn't figure out how to make a toolbar achieve that "docked" layout.) I started on the search feature, reading about [CollectionView](https://docs.microsoft.com/en-us/dotnet/desktop/wpf/controls/how-to-group-sort-and-filter-data-in-the-datagrid-control?view=netframeworkdesktop-4.8) and I added a [FilesCollection](https://github.com/adamfoneil/MyMusic.Wpf/blob/feature/search/MyMusic.Wpf/Models/Mp3View.cs#L26) property to the view model, but I couldn't get the grid to bind to it. I got a binding failure warning, but the warning itself had no information. Very weird.
